CMakeLists.txt

```diff
@@ -18,6 +18,9 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g")
 set(CMAKE_CXX_FLAGS_RELEASE "-Ofast -fno-finite-math-only")
 set(CMAKE_CXX_FLAGS_DEBUG  "-Wall -Wno-unused-local-typedefs -g")
 
+#Flag to avoid error in compilation
+add_compile_options(-fpermissive)
```

Comment:
I am curious, which compile error are you observing?

Reply:
When I was compiling on a base ubuntu image (the one for dockers), after installing all dependencies and  running cmake, the make command failed with the following console log:

```[ 25%] Built target fast_methods
Consolidate compiler generated dependencies of target fm_benchmark
[ 31%] Building CXX object CMakeFiles/fm_benchmark.dir/src/benchmark/fm_benchmark.cpp.o
In file included from /planning/include/fast_methods/fm/fmm.hpp:53,
                 from /planning/src/benchmark/fm_benchmark.cpp:25:
/planning/include/fast_methods/fm/eikonalsolver.hpp: In member function 'virtual double EikonalSolver<grid_t>::solveEikonal(const int&)':
/planning/include/fast_methods/fm/eikonalsolver.hpp:55:22: error: there are no arguments to 'isinf' that depend on a template parameter, so a declaration of 'isinf' must be available [-fpermissive]
   55 |                 if (!isinf(minTInDim) && minTInDim < grid_->getCell(idx).getArrivalTime())
      |                      ^~~~~
/planning/include/fast_methods/fm/eikonalsolver.hpp:55:22: note: (if you use '-fpermissive', G++ will accept your code, but allowing the use of an undeclared name is deprecated)
In file included from /usr/include/boost/config/header_deprecated.hpp:18,
                 from /usr/include/boost/progress.hpp:22,
                 from /planning/include/fast_methods/benchmark/benchmark.hpp:27,
                 from /planning/src/benchmark/fm_benchmark.cpp:34:
/usr/include/boost/progress.hpp: At global scope:
/usr/include/boost/progress.hpp:23:1: note: '#pragma message: This header is deprecated. Use the facilities in <boost/timer/timer.hpp> or <boost/timer/progress_display.hpp> instead.'
   23 | BOOST_HEADER_DEPRECATED( "the facilities in <boost/timer/timer.hpp> or <boost/timer/progress_display.hpp>" )
      | ^~~~~~~~~~~~~~~~~~~~~~~
/usr/include/boost/timer.hpp:21:1: note: '#pragma message: This header is deprecated. Use the facilities in <boost/timer/timer.hpp> instead.'
   21 | BOOST_HEADER_DEPRECATED( "the facilities in <boost/timer/timer.hpp>" )
      | ^~~~~~~~~~~~~~~~~~~~~~~
In file included from /planning/include/fast_methods/fm/fmm.hpp:53,
                 from /planning/src/benchmark/fm_benchmark.cpp:25:
/planning/include/fast_methods/fm/eikonalsolver.hpp: In instantiation of 'double EikonalSolver<grid_t>::solveEikonal(const int&) [with grid_t = nDGridMap<FMCell, 3>]':
/planning/include/fast_methods/fm/eikonalsolver.hpp:48:24:   required from here
/planning/include/fast_methods/fm/eikonalsolver.hpp:55:27: error: 'isinf' was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
   55 |                 if (!isinf(minTInDim) && minTInDim < grid_->getCell(idx).getArrivalTime())
      |                      ~~~~~^~~~~~~~~~~
In file included from /planning/src/benchmark/fm_benchmark.cpp:16:
/usr/include/c++/11/cmath:584:3: note: 'constexpr bool std::isinf(float)' declared here, later in the translation unit
  584 |   isinf(float __x)
      |   ^~~~~
In file included from /planning/include/fast_methods/fm/fmm.hpp:53,
                 from /planning/src/benchmark/fm_benchmark.cpp:25:
/planning/include/fast_methods/fm/eikonalsolver.hpp: In instantiation of 'double EikonalSolver<grid_t>::solveEikonal(const int&) [with grid_t = nDGridMap<FMCell, 2>]':
/planning/include/fast_methods/fm/eikonalsolver.hpp:48:24:   required from here
/planning/include/fast_methods/fm/eikonalsolver.hpp:55:27: error: 'isinf' was not declared in this scope, and no declarations were found by argument-dependent lookup at the point of instantiation [-fpermissive]
   55 |                 if (!isinf(minTInDim) && minTInDim < grid_->getCell(idx).getArrivalTime())
      |                      ~~~~~^~~~~~~~~~~
In file included from /planning/src/benchmark/fm_benchmark.cpp:16:
/usr/include/c++/11/cmath:584:3: note: 'constexpr bool std::isinf(float)' declared here, later in the translation unit
  584 |   isinf(float __x)
      |   ^~~~~
make[2]: *** [CMakeFiles/fm_benchmark.dir/build.make:76: CMakeFiles/fm_benchmark.dir/src/benchmark/fm_benchmark.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:137: CMakeFiles/fm_benchmark.dir/all] Error 2
make: *** [Makefile:136: all] Error 2 
```


As stated in the error, adding

```
add_compile_options(-fpermissive)
```
solves the error and compiles it successfully.

Just if you want to check  I used the following commands:

```
docker run -it --rm ubuntu

apt-get update
apt-get install build-essential -y
apt-get install nano cmake git doxygen cmake-curses-gui wget -y
apt-get install imagemagick libboost-all-dev cimg-dev  #requires user input

mkdir /planning
git clone https://github.com/jvgomez/fast_methods.git /planning
cd planning
mkdir build
cmake -S /planning/ -B /planning/build/
cd build
make
```

Reply:
Thanks for the info.  I am a bit worried about adding -fpermissive as it mostly hides the error, and doesn't solve it (https://stackoverflow.com/questions/8843818/what-does-the-fpermissive-flag-do).

Could you please try replacing all the isinf with std::isinf instead?